### ISSUE TITLE
Modifying ng-options list fails to render in IE 10 (and 9) when contained in transcluded content

### ISSUE THREAD

**Issue by ekaufman** [angular/angular.js#2809]
If you use a SELECT with ng-options underneath any directive that performs transclusion, and then modify the list used by ng-options, IE 9-10 will not automatically redraw the SELECT box to reflect the updated options unless the size of the list also happens to change.

Example:
```
    <div ng-switch on="dummy">
        <div ng-switch-default>
            My List: <select ng-model="listSelected" 
                       ng-options="item.value as item.name for item in childList">
                     </select>
        </div>
    </div>
```
If I swap out childList for another list of equal length, the browser still shows the previous options in the select until I click on it (and in IE9 even after clicking on it).

This fiddle demonstrates the issue in an ng-switch, and also in a custom-directive, and shows how when not in directive the problem doesn't occur.  The parent list drives changes to the child list, in IE you'll see that the child list doesn't redraw with the new options except in the third example which is the one that doesn't transclude.  Other browsers work as expected.
http://jsfiddle.net/redwood7/BGq3s/embedded/result/

Here's what I've tracked down so far; there is a weird side effect when you cloneNode() a SELECT in IE.  IE won't redraw the cloned SELECT unless you either add/remove options from it, or hide()/show() it, or click it, or do something else that forces it to redraw.  This fiddle demonstrates this problem completely outside of angular:
http://jsfiddle.net/redwood7/kHWxL/embedded/result/

The reason this relates to transclude in Angular is because directives that transclude use cloneNode() to copy the transcluded content. So, a SELECT with ng-options in that transcluded content end up having the problem.  And ng-options re-uses existing OPTIONs instead of removing and recreating them, so if your lists are equal length IE never redraws.

One workaround is to have a custom directive that inserts/removes a dummy option when IE.  Another is to ng-repeat to create options.  I'm hoping this is something that can be fixed in ng-options. Perhaps in IE make it recreate the OPTIONs instead of re-using them.

I've also posted this topic to the mailing list with some additional fiddles showing workarounds:
https://groups.google.com/forum/#!topic/angular/20SJvpwpt2o/discussion

**Comment by petebacondarwin** [comment]
This is a related article: https://brooknovak.wordpress.com/2009/08/23/ies-clonenode-doesnt-actually-clone/

**Comment by petebacondarwin** [comment]
@mzgol you are a master of this sort of thing. Do you have any ideas what we can do here?

**Comment by petebacondarwin** [comment]
I have moved it 1.4.x but if the fix is possible we could try to backport it to 1.3.x

**Comment by evilaliv3** [comment]
we are experiencing exactly the same issue on [GlobaLeaks](https://github.com/globaleaks/GlobaLeaks)!

**Comment by mgol** [comment]
The jsFiddle doesn't work for me in IE so I recreated this locally (esp. that the global controller declaration style is no longer supported) but I can't reproduce it with 1.5.0-rc.2, either on IE 10 or IE 9 on Windows 7 (checked via BrowserStack):
http://jsbin.com/koruxi/. I've also tried with Angular 1.4.9 & 1.3.20 on IE 9 and it works as well.

What I did: I changed the "Parent List" select values from A to B, in all 3 cases the child lists updated.

Does anyone have a full reproducible test case? Put it on jsbin or plunker, not jsfiddle as jsfiddle seems broken in IE.

**Comment by mgol** [comment]
What we've encountered in jQuery is that IE (all versions, fixed in Edge) don't always respect setting the `selected` property on an option so we have custom hooks that just access the `selectedIndex` property to make it behave:
https://github.com/jquery/jquery/blob/9cb89bf91d034ec7166d9215e2f80fa765292975/src/attributes/prop.js#L83-L109

It might be that this selecting fixes also some other issues, we can try sth like that. But I need a reproducible test case first. :)

Note, though, that we've had problems with IE 11 as well so if this problem appears only in IE<11 it might be unrelated.

**Comment by mgol** [comment]
> This is a related article: https://brooknovak.wordpress.com/2009/08/23/ies-clonenode-doesnt-actually-clone/

@petebacondarwin This is a description of just one tiny problem, IE has loads of them, especially versions older than 9. This particular problem has been fixed in IE 9 so it's not an issue for us.

**Comment by ekaufman** [comment]
The fix put in for https://github.com/angular/angular.js/issues/9621 has fixed this issue as well.

Here's the bug reproducible in 1.3.2:
http://jsbin.com/dezemuvune

And fixed in 1.3.3:
http://jsbin.com/deviwezequ

So it's fixed in 1.3.3+ see https://github.com/angular/angular.js/pull/10042
and also 1.2.27+ https://github.com/angular/angular.js/commit/16833d0fb6585117e9978d1accc3ade83e22e797

**Comment by ninad-d** [comment]
Hey I am facing somewhat related issue on IE11. I have two multiple select boxes and I select element from one select box and then via click of a button add it to second select box and vice versa.
The said works fine on all the browsers except IE11. On IE11 when I click on element from first select box and transfer it to another box it works but the first select list box goes unresponsive after that no amount of clicking or anything will select an element. 
`
        <select multiple="true" id="list-one" ng-model="vm.selectedItemsForFirstBox" ng-options="item as item.label for item in vm.items | filter:vm.filterText"></select>

<select multiple="true" ng-model="vm.selectedItemsForSecondBox" ng-options="item as item.label for item in vm.selectedItems"></select>`

So instead of the above I tried with ng-repeat, though ng-repeat solves the problem there rises another one. I also have a 'select all' button on click of which I select all the elements from a select and transfer it to another. Though the code works internally the select view does not gives me any indication of all elements being selected, in earlier code it use to highlight all elements with blue color that does not happens with this one. There is no visual indication to user that all elements have been selected. Code for same is,
`<select ng-if="vm.items" multiple="true" ng-model="vm.selectedItemsForFirstBox">
          <option ng-repeat="item in vm.items | filter:vm.filterText"
            value="{{item}}">
            {{item.label}}
          </option>
        </select> 
<select ng-if="vm.selectedItems" multiple="true" ng-model="vm.selectedItemsForSecondBox">
          <option ng-repeat="item in vm.selectedItems  | filter:vm.filterText"
            value="{{item}}">
            {{item.label}}
          </option>
        </select>`

I also tried the solution adding and removing a class from the list that also does not seems to work.
I am using Angular v1.4.9. Any of you guys aware about such issue? Or can point to me where I am going wrong? Tagging @ekaufman @mgol since you guys seem to know what might be culprit.
